lib/pages/rendezvous/rendezvous_list_page.dart

```diff
@@ -32,12 +32,13 @@ class _RendezvousListPageState extends State<RendezvousListPage> {
   Widget build(BuildContext context) {
     return StoreConnector<AppState, RendezvousListViewModel>(
       onInit: (store) {
-        if (_pageOffset == 0) store.dispatch(RendezvousRequestAction());
+        if (_pageOffset == 0) store.dispatch(RendezvousRequestAction(RendezvousPeriod.FUTUR));
```

Comment:
Il faudrait pas le dispatch du rendez vous past action machin chose ici du coup ?

Reply:
A l'origine, je voulais le mettre ici, sur offset -1, mais j'ai remarqué que l'init n'était jamais rappelé, donc j'ai dû faire la mécanique avec la closure onOffsetChanged. (ce qui m'amené à ma question du pourquoi vérifier offset 0 si ce n'est appelé qu'une seule fois)

Reply:
C'est étrange que ce ne soit jamais appelé. Quand je l'avais implémenté la première fois c'était appelé. Mais oui si c'est jamais appelé tu peux l'enlever

Reply:
J'ai eu le même retout @RemiDormoy :) avant de comprendre que c'est fait dans le viewmodel

Reply:
Par contre, je me demande si ce n'est pas la condition `if (_pageOffset == 0) ` qui est trompeuse. Que penses-tu de la supprimer, sachant qu'on arrive toujours sur l'offset 0, non ?

Reply:
Mais je suis d'accord, j'ai du faire des recherches pour le comprendre. Pour le coup, je me demande si un petit commentaire de genre `// retrieval of past rendezvous is directly made in ViewModel` pourrait avoir du sens ?
Ou même, toujours tout faire dans le ViewModel ? Pourquoi pas avec une méthode du ViewModel fetchRendezvous, qui prendrait le page offset en param. Cette méthode ne ferait pas l'appel réseau si le state passé ou futur est déjà plein. Et elle pourrait-être appelée dans le `converter` de la `RendezvousListPage` juste avant l'appel à `RendezvousListViewModel.create(store, DateTime.now(), _pageOffset)` ? 
Ca fait pas mal de code en plus, pour le bénéfice de n'avoir qu'un point d'entrée. Donc pas super sûr de cette proposition. Qu'en penses-tu ?

Reply:
Ma compréhension actuelle flutter+redux me fait dire que c'est bien que la première action soit lancée dans le `onInit`, c'est un bon endroit conseillé pour alimenter en données un widget. Et c'est fonctionnellement ce que l'on veut : on lance uniquement le futur à l'arrivée sur la page.
Je n'ai pas réussi à trouver d'explications sur le pourquoi il y a eu besoin de mettre le `if offset == 0`, on peut le retirer si on est serein, avec une explication c'est bien, ou de retrouver le commit qui l'a introduit pour voir.
Mettre tout (les actions de requêtes) au même endroit, dans l'idée, j'aimerais bien aussi. Mais, quel est cet endroit ? Il y a peu d'endroit avec le store au final. `onInit` ? Eliminé, pas appelé au changement de page. `converter` ? A chaud je dirais non, sa responsabilité, c'est de _mapper_ un état vers une modélisation de vue (vm), pas de faire de l'effet de bord.

Reply:
Si on fait en sorte que le `onInit` soit appelé aux changements de pages, ce serait la solution idéale je pense. Il appellerait simplement le VM pour lui dire "hey, on change d'offset, peut-être que tu dois charger des données". Et toute la logique serait dans cette méthode du VM, pour le futur et le passé. Mais arrive-t-on à déclencher le `onInit` ? Petit défi de demain :)

Reply:
OK, merci pour tes retours :) J'ai pushé / mergé en enlevant la conditions qui n'a plus de sens depuis que tu as mis le onDispose qui reset le state. Avec un commentaire en plus.
Et je vais tester sur une autre branche ce que j'avais en tête. Mais le `onInit` ne peut pas être appelé 2 fois, à moins que l'on fasse 2 Widgets différents, et ce n'est pas ce que l'on veut.